Automattic/wp-calypso
# Expose Hosting Trial: Show free trial CTA on plugins upload page for free sites

Related to https://github.com/Automattic/dotcom-forge/issues/5097

## Proposed Changes
This PR allow free site users to start a free trial from the plugins/upload page by showing the a contextual CTA "Start your free trial". This CTA is only shown if the site is free and the user did not have a free trial before. 

* Show "Start your free trial" CTA 
* add trial plan 
## Testing Instructions



* go to` /plugins/upload/free_site` verify the CTA is shown
* Verify redirect to checkout with the free trial plan
* 
![image](https://github.com/Automattic/wp-calypso/assets/47489215/f55703b4-9f9b-4eab-b4dd-70aa2a339820)


## Pre-merge Checklist
